RIRImageResizing/Classes/NSObject/RIRResizeImageOperation.m

```diff
@@ -0,0 +1,189 @@
+//
+//
+
+#import "RIRResizeImageOperation.h"
+
+#import <ResplendentUtilities/RUConditionalReturn.h>
+
+
+
+
+
+@interface RIRResizeImageOperation ()
+
+#pragma mark - scale
+/**
+ Returns a new `UIImage` instance, which is a resized version of the receiver.
+ 
+ @param newSize    The size that the new UIImage instance aims for. This is not necessarily the size of the final image, as it depends on the target image's size, and the `resizeMode` parameter.
+ @param resizeMode Indicates which type of resize method to use.
+ 
+ @return A new `UIImage` instance, which is a resized version of the receiver.
+ */
+-(nullable UIImage*)scaleToSize:(CGSize)newSize usingMode:(UIImage_RIRResizing_ResizeMode)resizeMode;
+
+/**
+ Returns a new `UIImage` instance, which is a resized version of the receiver. Uses `UIImage_RIRResizing_ResizeMode_aspectFit` resizing mode.
+ 
+ @param newSize    The size that the new UIImage instance aims for. This is not necessarily the size of the final image, as it depends on the target image's size, and the `resizeMode` parameter.
+ 
+ @return A new `UIImage` instance, which is a resized version of the receiver.
+ */
+-(nonnull UIImage*)scaleToFitSize:(CGSize)newSize;		//UIImage_RIRResizing_ResizeMode_aspectFit
+
+/**
+ Returns a new `UIImage` instance, which is a resized version of the receiver. Uses `UIImage_RIRResizing_ResizeMode_scaleToFill` resizing mode.
+ 
+ @param newSize    The size that the new UIImage instance aims for. This is not necessarily the size of the final image, as it depends on the target image's size, and the `resizeMode` parameter.
+ 
+ @return A new `UIImage` instance, which is a resized version of the receiver.
+ */
+-(nonnull UIImage*)scaleToFillSize:(CGSize)newSize;		//UIImage_RIRResizing_ResizeMode_scaleToFill
+
+/**
+ Returns a new `UIImage` instance, which is a resized version of the receiver. Uses `UIImage_RIRResizing_ResizeMode_aspectFill` resizing mode.
+ 
+ @param newSize    The size that the new UIImage instance aims for. This is not necessarily the size of the final image, as it depends on the target image's size, and the `resizeMode` parameter.
+ 
+ @return A new `UIImage` instance, which is a resized version of the receiver.
+ */
+-(nonnull UIImage*)scaleToCoverSize:(CGSize)newSize;	//UIImage_RIRResizing_ResizeMode_aspectFill
+
+@end
+
+
+
+
+
+@implementation RIRResizeImageOperation
+
+#pragma mark - NSObject
+-(instancetype)init
+{
+    kRUConditionalReturn_ReturnValueNil(YES, YES);
+    return [self init_with_resizeParameters:nil
+                                      image:nil];
+}
+
+#pragma mark - init
+-(instancetype)init_with_resizeParameters:(RIRResizeImageOperationParameters *)resizeParameters
+                                    image:(UIImage *)image
+{
+    kRUConditionalReturn_ReturnValueNil(resizeParameters == nil, YES);
+    kRUConditionalReturn_ReturnValueNil(image == nil, YES);
+    
+    if (self = [super init])
+    {
+        _resizeParameters = resizeParameters;
```

Comment:
We generally want to use setter methods instead of instance variables, in case we one day wanted to be able to KVO on these properties (and other reasons, it's good convention).

Go ahead and declare these properties in the .m's class extension, as they're declared in the .h, but without `readonly`, then call the setters here.